src/AppBundle/Templating/Helper/SourceCodeHelper.php

```diff
@@ -0,0 +1,129 @@
+<?php
+/**
+ * This file is part of the Symfony package.
+ *
+ * @copyright (c) 2015, symfony-demo
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace AppBundle\Templating\Helper;
+
+use ReflectionClass;
+use Symfony\Component\Templating\EngineInterface;
+use Symfony\Component\Templating\Helper\Helper;
+
+/**
+ * Class SourceCodeHelper
+ *
+ * @author Rasanga Perera <rasangaperera@gmail.com>
+ */
+class SourceCodeHelper extends Helper
+{
+    /**
+     * @var EngineInterface
+     */
+    private $templating;
+
+    /**
+     * @var string
+     */
+    private $kernelRootDir;
+
+    /**
+     * @param EngineInterface $templating
+     * @param string $kernelRootDir
+     */
+    public function __construct(
+        EngineInterface $templating,
+        $kernelRootDir
+    ) {
+        $this->templating = $templating;
+        $this->kernelRootDir = $kernelRootDir;
+    }
+
+    /**
+     * @param string $templateName
+     * @param string $templateSource
+     * @param array|null $controller
+     *
+     * @return string
+     */
+    public function showSourceCode(
+        $templateName,
+        $templateSource,
+        $controller = null
+    ) {
+        return $this->templating->render('default/_source_code.html.twig', array(
```

Comment:
if you were really writing a templating helper for the PHP engine, making it use a Twig templating for its implementation would be a very bad idea, as it would force to enable Twig to be able to use the PHP template helper